src/main/kotlin/club/sk1er/vigilance/data/SortingBehavior.kt

```diff
@@ -0,0 +1,20 @@
+package club.sk1er.vigilance.data
+
+open class SortingBehavior {
+    /**
+     * @return a comparator, that is used to sort a list of categories
+     */
+    open fun getCategoryComparator(): Comparator<in Category> = compareBy { it.name }
+
+    /**
+     * @return a comparator, that is used to sort a map of subcategories
+     * Key of an entry = name of the subcategory
+     * Value of an entry = list of properties in the subcategory
+     */
+    open fun getSubcategoryComparator(): Comparator<in Map.Entry<String, List<PropertyData>>> = compareBy { null }
```

Comment:
Should we simplify this to `Comparator<in String>` or do you believe the subcategory comparator actually need the list of properties it contains? Open to opinions.

Reply:
Majority of comparators probably are fine with just the name.
My thought was that if you can provide more somewhat relevant information to the comparator, provide it. It allows more uses without taking anything (except maybe some amount of simplicity) away.
I am fine if you want to simplify it.

Reply:
Well, I understand trying to allow it to be as powerful as we can offer, but is there actually a realistic scenario in which it would be useful? If it theoretically might be, we can leave it in.

Reply:
Only one that I can quickly think of while being something that someone could use is sorting subcategories by the number of properties.

Reply:
Fair enough, we can leave it. The default behavior for subcategory sorting should be sorting by subcategory name though, no?

Reply:
Correct, it **should** be sorted by the subcategory name. I'll fix that.